File: src/test/kotlin/io/miiken/intellijcontrolserver/fixtures/scala/SampleClass.scala
package testdata.scala

/**
 * Sample Scala class for testing refactoring operations.
 * 
 * This class includes test cases for renaming:
 * - Methods
 * - Fields
 * - Parameters
 * - Local variables
 * - Classes
 */
class SampleCalculator {
  // RENAME THIS - Test renaming this field
  private var oldFieldName: Int = 10
  
  // RENAME THIS - Test renaming this method
  def oldMethodName(oldParameterName: Int): Int = {
    // RENAME THIS - Test renaming this local variable
    val oldVariableName = oldParameterName * 2
    oldVariableName + oldFieldName
  }
  
  /**
   * Calculate discount based on total amount.
   * Calls oldMethodName to apply bonus points.
   */
  def calculateDiscount(total: Double, membershipLevel: String): Double = {
    val baseDiscount = membershipLevel match {
      case "gold" => 0.20
      case "silver" => 0.10
      case "bronze" => 0.05
      case _ => 0.0
    }
    
    val volumeDiscount = if (total > 100.0) 0.05 else 0.0
    
    // Call oldMethodName to calculate bonus points
    val bonusPoints = oldMethodName(total.toInt)
    oldFieldName += bonusPoints
    
    Math.min(baseDiscount + volumeDiscount, 0.30)
  }
  
  /**
   * Process a purchase.
   * Calls calculateDiscount and oldMethodName.
   */
  def processPurchase(amount: Double, level: String): Double = {
    val discount = calculateDiscount(amount, level)
    val finalAmount = amount * (1.0 - discount)
    
    // Update oldFieldName through oldMethodName
    oldMethodName(amount.toInt)
    
    finalAmount
  }
}

File: src/test/kotlin/io/miiken/intellijcontrolserver/fixtures/typescript/SampleClass.ts
/**
 * Sample TypeScript class for testing refactoring operations.
 * 
 * This class includes test cases for renaming:
 * - Methods
 * - Fields
 * - Parameters
 * - Local variables
 * - Classes
 */
class SampleCalculator {
    // RENAME THIS - Test renaming this field
    private oldFieldName: number = 10;
    
    // RENAME THIS - Test renaming this method
    oldMethodName(oldParameterName: number): number {
        // RENAME THIS - Test renaming this local variable
        const oldVariableName = oldParameterName * 2;
        return oldVariableName + this.oldFieldName;
    }
    
    /**
     * Calculate discount based on total amount.
     * Calls oldMethodName to apply bonus points.
     */
    calculateDiscount(total: number, membershipLevel: string): number {
        let baseDiscount: number;
        switch (membershipLevel) {
            case 'gold':
                baseDiscount = 0.20;
                break;
            case 'silver':
                baseDiscount = 0.10;
                break;
            case 'bronze':
                baseDiscount = 0.05;
                break;
            default:
                baseDiscount = 0.0;
        }
        
        const volumeDiscount = total > 100.0 ? 0.05 : 0.0;
        
        // Call oldMethodName to calculate bonus points
        const bonusPoints = this.oldMethodName(Math.floor(total));
        this.oldFieldName += bonusPoints;
        
        return Math.min(baseDiscount + volumeDiscount, 0.30);
    }
    
    /**
     * Process a purchase.
     * Calls calculateDiscount and oldMethodName.
     */
    processPurchase(amount: number, level: string): number {
        const discount = this.calculateDiscount(amount, level);
        const finalAmount = amount * (1.0 - discount);
        
        // Update oldFieldName through oldMethodName
        this.oldMethodName(Math.floor(amount));
        
        return finalAmount;
    }
}

export default SampleCalculator;
